File: examples/nodejs/twitch.js
const axios = require('axios');

const API_BASE = 'https://cog.api.br/api/v1';

/**
 * Download Twitch clip or VOD
 * @param {string} url - Twitch clip or VOD URL
 */
async function downloadTwitch(url) {
  try {
    const response = await axios.get(`${API_BASE}/twitch/download`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Twitch Video Downloaded Successfully!');
      console.log('Title:', response.data.data.title);
      console.log('Streamer:', response.data.data.streamer);
      console.log('Type:', response.data.data.type);
      console.log('Game:', response.data.data.game);
      console.log('Views:', response.data.data.views);
      console.log('Download URL:', response.data.data.downloadUrl);
      console.log('Duration:', response.data.data.duration, 'seconds');
      
      return response.data.data;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get available formats for Twitch video
 * @param {string} url - Twitch clip or VOD URL
 */
async function getTwitchFormats(url) {
  try {
    const response = await axios.get(`${API_BASE}/twitch/formats`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Available Formats:');
      response.data.formats.forEach((format, index) => {
        console.log(`\nFormat ${index + 1}:`);
        console.log('  ID:', format.formatId);
        console.log('  Quality:', format.quality);
        console.log('  Extension:', format.ext);
        console.log('  Size:', (format.filesize / 1024 / 1024).toFixed(2), 'MB');
      });
      
      return response.data.formats;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get information about Twitch clip or VOD
 * @param {string} url - Twitch clip or VOD URL
 */
async function getTwitchInfo(url) {
  try {
    const response = await axios.get(`${API_BASE}/twitch/info`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Twitch Video Information:');
      console.log('Title:', response.data.info.title);
      console.log('Streamer:', response.data.info.streamer);
      console.log('Type:', response.data.info.type);
      console.log('Game:', response.data.info.game);
      console.log('Views:', response.data.info.views);
      console.log('Duration:', response.data.info.duration, 'seconds');
      console.log('Description:', response.data.info.description);
      
      return response.data.info;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

// Example usage
(async () => {
  console.log('=== Twitch Download Example ===\n');
  
  // Example 1: Download Twitch clip
  const clipUrl = 'https://clips.twitch.tv/example';
  console.log('Downloading Twitch clip:', clipUrl);
  await downloadTwitch(clipUrl);
  
  console.log('\n=== Twitch Formats Example ===\n');
  
  // Example 2: Get available formats
  const vodUrl = 'https://www.twitch.tv/videos/123456789';
  console.log('Getting formats for:', vodUrl);
  await getTwitchFormats(vodUrl);
  
  console.log('\n=== Twitch Info Example ===\n');
  
  // Example 3: Get video info
  console.log('Getting video info:', clipUrl);
  await getTwitchInfo(clipUrl);
})();

module.exports = {
  downloadTwitch,
  getTwitchFormats,
  getTwitchInfo
};

File: examples/python/twitch.py
import requests

API_BASE = 'https://cog.api.br/api/v1'

def download_twitch(url: str) -> dict:
    """
    Download Twitch clip or VOD
    
    Args:
        url: Twitch clip or VOD URL
        
    Returns:
        dict: Download data or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/twitch/download', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Twitch Video Downloaded Successfully!')
            print(f"Title: {data['data']['title']}")
            print(f"Streamer: {data['data']['streamer']}")
            print(f"Type: {data['data']['type']}")
            print(f"Game: {data['data']['game']}")
            print(f"Views: {data['data']['views']}")
            print(f"Download URL: {data['data']['downloadUrl']}")
            print(f"Duration: {data['data']['duration']} seconds")
            
            return data['data']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_twitch_formats(url: str) -> list:
    """
    Get available formats for Twitch video
    
    Args:
        url: Twitch clip or VOD URL
        
    Returns:
        list: Available formats or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/twitch/formats', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Available Formats:')
            for index, format_info in enumerate(data['formats'], 1):
                print(f'\nFormat {index}:')
                print(f"  ID: {format_info['formatId']}")
                print(f"  Quality: {format_info['quality']}")
                print(f"  Extension: {format_info['ext']}")
                print(f"  Size: {format_info['filesize'] / 1024 / 1024:.2f} MB")
            
            return data['formats']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_twitch_info(url: str) -> dict:
    """
    Get information about Twitch clip or VOD
    
    Args:
        url: Twitch clip or VOD URL
        
    Returns:
        dict: Video information or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/twitch/info', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Twitch Video Information:')
            print(f"Title: {data['info']['title']}")
            print(f"Streamer: {data['info']['streamer']}")
            print(f"Type: {data['info']['type']}")
            print(f"Game: {data['info']['game']}")
            print(f"Views: {data['info']['views']}")
            print(f"Duration: {data['info']['duration']} seconds")
            print(f"Description: {data['info']['description']}")
            
            return data['info']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

if __name__ == '__main__':
    print('=== Twitch Download Example ===\n')
    
    # Example 1: Download Twitch clip
    clip_url = 'https://clips.twitch.tv/example'
    print(f'Downloading Twitch clip: {clip_url}')
    download_twitch(clip_url)
    
    print('\n=== Twitch Formats Example ===\n')
    
    # Example 2: Get available formats
    vod_url = 'https://www.twitch.tv/videos/123456789'
    print(f'Getting formats for: {vod_url}')
    get_twitch_formats(vod_url)
    
    print('\n=== Twitch Info Example ===\n')
    
    # Example 3: Get video info
    print(f'Getting video info: {clip_url}')
    get_twitch_info(clip_url)
